### DISCUSSION TITLE
mpv 39 opens certain files much slower than 38

### DISCUSSION THREAD

**Discussion by Nyaa-rlathotep** [mpv-player/mpv#15606] [Q&A] [answered]
For some reason, certain files (it seems to mostly be ones with a static image with audio, rather than proper video) will take a long time to open depending on the file size or length. Short files take about 5-10 seconds to open with mpv 39, but a multiple hour long one I have takes at least several minutes, though I manually killed mpv after waiting that long.

Mpv 38 opens all of these files pretty much instantly.

Running mpv with the --no-config option doesn't seem to make a difference.

Thankfully, it's fully reproducible on my system, so testing is easy. I expect I probably just have a build of mpv that's maybe accidentally missing something important, but I'm not sure what, since I havn't changed the build options between 38 and 39.

If it makes a difference, I'm running a system with AMD integrated graphics. Specifically, the AMD 7840U chip.

I'm attaching example files that open quick and slow in 39, and I'll list the build options below. (I had to zip the files, as github apparently doesn't like mkv files.)

[example_files.zip](https://github.com/user-attachments/files/18282536/example_files.zip)

Note that the below build options are Gentoo "USE flags", so they might be named slightly differently from the actual build config settings, but they should help in case I've mistakenly left out an important feature or anything.
Included and excluded build features:

Included:
alsa
bluray
cli
drm
egl
iconv
libmpv
lua
pipewire
pulseaudio
uchardet
vulkan
wayland
zlib

Excluded:
aqua
archive
cdda
coreaudio
debug
dvb
dvd
gamepad
jack
javascript
jpeg
lcms
libcaca
nvenc
openal
opengl
rubberband
sdl
selinux
sixel
sndio
soc
test
tools
vaapi
vdpau
xv
zimg

**Comment by llyyr** [comment]
Bisected to https://github.com/mpv-player/mpv/commit/22ed5d0d3a2e1db99f5e2d6165f0ecd013fef4d1

You can workaround this issue by using `attached_picture` when making your mkv file instead. In fact that's no workaround at all, and your files as they are is currently broken and mpv automagically detects that your "video" track is an image.

cc @kasper93

**Reply by Dudemanguy** [reply]
We later changed the probe down to just 10000 blocks in d54e8c3f40934501d5fb5875defd797eff0ae57b. This was after v0.39 but presumably that should fix any speed issues given it's a whole magnitude less probing.

**Comment by Nyaa-rlathotep** [comment]
Unfortunately "attached_pic" doesn't seem to make a difference, unless maybe I'm doing something wrong.

Comand line:
ffmpeg -i temp.opus -i temp.png -map 0 -map 1 -codec copy -disposition:v:1 attached_pic "slow_open.mkv"

(Edit: ffmpeg is version 6.1.2)

The command is taken from https://ffmpeg.org/ffmpeg.html. I also tried setting the disposition to "attached_picture" as well, with no change.
Manually specifying to use the png codec (with "-c:v:1 png") also made no difference.


[slow_open-attached_pic.zip](https://github.com/user-attachments/files/18282658/slow_open-attached_pic.zip)

**Reply by guidocella** [reply, marked as answer]
https://video.stackexchange.com/questions/35598/is-it-possible-to-embed-a-thumbnail-in-a-mkv-file

**Reply by Nyaa-rlathotep** [reply]
Thanks! Using `-attach temp.png -metadata:s:t:0 mimetype=image/png` to attach the file instead of -i gets it working quickly in mpv again.

Anyone know if there's a way to avoid issues like this in the future? ffmpeg and mpv never complained about the previous method of including a still image as the "video", so it appeared to be a totally valid way of doing things at the time.
